eclass/ninja-utils.eclass

```diff
@@ -55,14 +60,64 @@ _NINJA_UTILS_ECLASS=1
 
 inherit multiprocessing
 
-NINJA_DEPEND="app-alternatives/ninja"
+case ${NINJA} in
+	ninja)
+		NINJA_DEPEND="app-alternatives/ninja"
+		;;
+	ninja-reference)
+		NINJA_DEPEND="dev-build/ninja"
+		;;
+	samu)
+		NINJA_DEPEND="dev-build/samurai"
+		;;
+esac
+
+# @FUNCTION: _ninja_uses_jobserver
+# @DESCRIPTION:
+# Return true if current ${NINJA} has jobserver support and we have one
+# running (via MAKEFLAGS).
+_ninja_uses_jobserver() {
+	# ninja supports jobserver via FIFO only
+	[[ ${MAKEFLAGS} == *--jobserver-auth=fifo:* ]] || return 1
+
+	case ${NINJA} in
+		# if using "ninja", make sure its a symlink to real ninja
```

Comment:
Add a link to https://github.com/michaelforney/samurai/issues/71 (sadly there's two PRs pending).